lib/system.nim

```diff
@@ -98,6 +98,17 @@ type
   SomeNumber* = SomeInteger|SomeReal
     ## type class matching all number types
 
+  T1* = auto
```

Comment:
These helpers are useful for writing implicitly generic procs as in:

``` Nim
proc find(s: seq[T1], value: T1)
```

We've discussed them a bit on IRC in the past and I managed to convince you back then that they belong to the system module. In general, the are not related to `typetraits` in any way.